### PR TITLE
Move job, models, and feature_slice_view plotting to API.

### PR DESCRIPTION
**Pull request by qimingj**

### CODE REVIEW

File: datalab/mlalpha/_cloud_models.py [hunk 1]
```diff
@@ -95,11 +89,36 @@ def delete(self, model_name):
...
+    Args:
+      count: upper limit of the number of models to list.
+    Raises:
+      Exception if it is called in a non-IPython environment.
+    """
+    import IPython
+    data = [{'name': model['name'], 
+             'defaultVersion': model['defaultVersion']['name'].split('/')[-1]
+                 if 'defaultVersion' in model else None}
+            for _, model in zip(range(count), self)]
```
Comment (brandondutra) [inline datalab/mlalpha/_cloud_models.py]: so model is always equal to self? This list comprehension is complicated enough to just do it in a loop.

File: datalab/mlalpha/_cloud_models.py [hunk 2]
```diff
@@ -95,11 +89,36 @@ def delete(self, model_name):
...
+    IPython.display.display(
+        datalab.utils.commands.render_dictionary(data, ['name', 'defaultVersion']))
+    
+  def describe(self, model_name):
+    """Print information of a specified model.
+
+    Args:
+      model_name: the name of the model to print details on.
+    """    
+    model_yaml = yaml.safe_dump(self.get(model_name), default_flow_style=False)
```
Comment (brandondutra) [inline datalab/mlalpha/_cloud_models.py]: can we change get() to a better name, or add a todo to do it? get() is void of meaning, and it looks like a dict's get().

File: datalab/mlalpha/_cloud_models.py [hunk 3]
```diff
@@ -95,11 +89,36 @@ def delete(self, model_name):
...
+    print model_yaml
+    
 
 class CloudModelVersions(object):
   """Represents a list of versions for a Cloud ML model."""
 
-  def __init__(self, model_name, project_id=None, credentials=None, api=None):
+  def __init__(self, model_name, project_id=None):
     """Initializes an instance of a CloudML model version list that is iteratable
         ("for version in CloudModelVersions()").
```
Comment (brandondutra) [inline datalab/mlalpha/_cloud_models.py]: I don't think CloudModelVersions is iterable, so the comment "for version in CloudModelVersions()" is wrong? Likewise for CloudModels.

File: datalab/mlalpha/_feature_slice_view.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    if len(df.columns) < 3:
+      raise Exception('Need at least one metrics column.')      
+    if len(df) == 0:
+      raise Exception('Data is empty')
+
+    metric_names = list(set(df) - set(['feature']))
+    data = []
+    for ii, row in df.iterrows():
+      metric_values = dict(row)
+      metric_values['count'] = metric_values['count']
```
Comment (brandondutra) [inline datalab/mlalpha/_feature_slice_view.py]: what is this doing?

File: datalab/mlalpha/_feature_slice_view.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    else:
+      raise Exception('data needs to be a sql query, or a pandas DataFrame.')
+      
+    HTML_TEMPLATE = """<link rel="import" href="/nbextensions/gcpdatalab/extern/lantern-browser.html" >
+        <lantern-browser id="%s"></lantern-browser>
+        <script>
+        var browser = document.querySelector('#%s');
+        browser.metrics = %s;
+        browser.data = %s;
+        browser.sourceType = 'colab';
```
Comment (brandondutra) [inline datalab/mlalpha/_feature_slice_view.py]: colab -> datalab?

File: datalab/mlalpha/_feature_slice_view.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        browser.weightedExamplesColumn = 'count';
+        browser.calibrationPlotUriFn = function(s) { return '/' + s; }
+        </script>"""
+    metrics_str = str(map(str, data[0]['metricValues'].keys()))
+    data_str = str([{str(k): json.dumps(v) for k,v in elem.iteritems()} for elem in data])
+    html_id = 'l' + datalab.utils.commands.Html.next_id()
+    html = HTML_TEMPLATE % (html_id, html_id, metrics_str, data_str)
```
Comment (brandondutra) [inline datalab/mlalpha/_feature_slice_view.py]: maybe use .format(html_id=xxx, ....)?

File: datalab/mlalpha/_cloud_models.py (hunk 1 above)
Comment (qimingj) [inline datalab/mlalpha/_cloud_models.py]: self is an iterable (see __iter__() method). I expanded it to a loop.

File: datalab/mlalpha/_cloud_models.py (hunk 2 above)
Comment (qimingj) [inline datalab/mlalpha/_cloud_models.py]: Changed it to get_model_details().

File: datalab/mlalpha/_cloud_models.py (hunk 3 above)
Comment (qimingj) [inline datalab/mlalpha/_cloud_models.py]: It is. See "__iter__()".

File: datalab/mlalpha/_feature_slice_view.py (hunk 4 above)
Comment (qimingj) [inline datalab/mlalpha/_feature_slice_view.py]: Added comments. Feature slice view browser expects the data in a certain format.

File: datalab/mlalpha/_feature_slice_view.py (hunk 5 above)
Comment (qimingj) [inline datalab/mlalpha/_feature_slice_view.py]: Unfortunately 'colab' is a hard coded value for the browser.

File: datalab/mlalpha/_feature_slice_view.py (hunk 6 above)
Comment (qimingj) [inline datalab/mlalpha/_feature_slice_view.py]: Done.

**Pull request merged**